### ISSUE TITLE
Compile error: 'EUDVariable' object has no attribute 'fields', when `a[0]` used as logical operand

### ISSUE THREAD

**Issue by Chromowolf** [armoha/euddraft#83]
For euddraft0970 and later versions:
```js
const a = PVariable();

function onPluginStart() {
    if (a[0]) {  // a[0]>0 also error
        // some statement
    }
}
```
Compile error:
[Error] 'EUDVariable' object has no attribute 'fields'

**Comment by armoha** [comment]
This should be fixed in euddraft 0.9.7.4.

But new issue remains while moving optimizing pass from eudplib to epScript.
In euddraft 0.9.7.4, `if (a[0])` will call subscript while `if (a[0] > 0)` does in-place comparison. :(

**Comment by armoha** [comment]
Fixed confirmed in euddraft 0.9.7.7
